lib/deserializers.php

```diff
@@ -161,3 +161,31 @@ function elementList(Reader $reader, $namespace = null) {
     return $values;
 
 }
+
+/**
+ * @param Reader $reader
+ * @param $popo
+ * @param $namespace
+ * @return null|$popo
+ */
+function popoObject(Reader $reader, $popo, $namespace) {
+    if ($reader->isEmptyElement) {
+        $reader->next();
+        return null;
+    }
+
+    $reader->read();
+    do {
+
+        if ($reader->nodeType === Reader::ELEMENT && $reader->namespaceURI == $namespace) {
```

Comment:
Its one indirection and another short living temp. Structure.. In case you prefer it I am fine with it. We can restore to what we have right now in case it get a perf bottleneck